devito/operator.py

```diff
@@ -82,7 +82,7 @@ def expr_cse(expr):
     for temp, value in temps:
         if isinstance(value, IndexedBase):
             to_revert[temp] = value
-        elif isinstance(value, Add):
+        elif isinstance(value, Add) and not set([t, x, y, z]).isdisjoint(set(value.args)):
```

Comment:
Ok, I'm fine with this temporary solution then. We'll need to do a lot of this unpicking anyway once we start using the `Dimension`/`Iteration` objects properly.